From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect enum enclosure_component_type
Patch-mainline: never, kabi

In 3.12.56, commit 107626c1d8e73311be1815837b16a2aa0195fdfe (ses: fix
additional element traversal bug), upstream commit
5e1033561da1152c57b97ee84371dba2b3d64c25 added few new values to enum
enclosure_component_type. It has no effect on the other values,
because they are numbered.

So only protect the new enum entries by #ifdef __GENKSYMS__.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/linux/enclosure.h |    4 ++++
 1 file changed, 4 insertions(+)

--- a/include/linux/enclosure.h
+++ b/include/linux/enclosure.h
@@ -29,11 +29,15 @@
 /* A few generic types ... taken from ses-2 */
 enum enclosure_component_type {
 	ENCLOSURE_COMPONENT_DEVICE = 0x01,
+#ifndef __GENKSYMS__
 	ENCLOSURE_COMPONENT_CONTROLLER_ELECTRONICS = 0x07,
 	ENCLOSURE_COMPONENT_SCSI_TARGET_PORT = 0x14,
 	ENCLOSURE_COMPONENT_SCSI_INITIATOR_PORT = 0x15,
+#endif
 	ENCLOSURE_COMPONENT_ARRAY_DEVICE = 0x17,
+#ifndef __GENKSYMS__
 	ENCLOSURE_COMPONENT_SAS_EXPANDER = 0x18,
+#endif
 };
 
 /* ses-2 common element status */
